src/shared/components/TruncatedTextSVG.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from "react";
+import {observer} from "mobx-react";
+import {action, computed, observable} from "mobx";
+import autobind from "autobind-decorator";
+import * as ReactDOM from "react-dom";
+import {Popover} from "react-bootstrap";
+import classnames from "classnames";
+import styles from "../../pages/resultsView/survival/styles.module.scss";
+import {TOOLTIP_MOUSE_ENTER_DELAY_MS} from "./defaultTooltip/DefaultTooltip";
+import {truncateWithEllipsis, truncateWithEllipsisReport} from "../lib/wrapText";
+
+export interface ITruncatedTextSVGProps {
+    text?:string;
+    maxWidth?:number;
+    suffix?:string;
+    tooltip?:(datum:any)=>JSX.Element;
+    tooltipPlacement?:string;
+    dy?:any;
+    //victory
+    data?:any;
+    datum?:any;
+    style?:{
+        fontSize:number;
+        fontFamily:string;
+    };
+}
+
+@observer
+export default class TruncatedTextSVG extends React.Component<ITruncatedTextSVGProps, {}> {
+
+    static defaultProps:Partial<ITruncatedTextSVGProps> = {
+        maxWidth:100,
+        suffix:"..."
+    };
+
+    @observable mousePosition = { x:0, y:0 };
+    @observable tooltipOpen = false;
+
+    @computed get tooltipElt() {
+        if (this.props.tooltip && this.props.datum) {
+            return this.props.tooltip(this.props.datum);
+        } else {
+            return <span>{this.props.text}</span>;
+        }
+    }
+
+    @autobind
+    @action private onMouseOver() {
+        this.tooltipOpen = true;
+    }
+
+    @autobind
+    @action private onMouseOut() {
+        this.tooltipOpen = false;
+    }
+
+    @autobind private onMouseMove(e:React.MouseEvent<any>) {
+        this.mousePosition.x = e.pageX;
+        this.mousePosition.y = e.pageY;
+    }
+
+    @computed get fontFamily() {
+        let font = "Arial";
+        if (this.props.style) {
+            font = this.props.style.fontFamily.split(", ")[0];
+        }
+        return font;
+    }
+
+    @computed get fontSize() {
+        let size = "13px";
+        if (this.props.style) {
+            size = this.props.style.fontSize + "px";
+        }
+        return size;
+    }
+
+    @computed get textReport() {
+        return truncateWithEllipsisReport(
+            this.props.text || "",
+            this.props.maxWidth!,
+            this.fontFamily,
+            this.fontSize
+        );
+    }
+
+    render() {
+        const {text, maxWidth, data, datum, suffix, tooltip,
+            tooltipPlacement, ...rest} = this.props;
+        return (
+            <>
+                <text
+                    {...rest}
+                    onMouseOver={this.onMouseOver}
+                    onMouseOut={this.onMouseOut}
+                    onMouseMove={this.onMouseMove}
+                >
+                    {this.textReport.text}
+                </text>
+                {this.textReport.truncated && this.tooltipOpen && (ReactDOM as any).createPortal(
```

Comment:
we can use react-portal library to avoid imperative call to creatPortal.  e.g. <Portal>

Reply:
okay cool